pkg/internal/approver/validation/validator.go

```diff
@@ -0,0 +1,85 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/ext"
+)
+
+const (
+	varSelf    = "self"
+	varRequest = "cr"
+)
+
+// Validator knows how to validate CSR attribute values in CertificateRequests
+// against CEL expressions declared in CertificateRequestPolicy.
+// Validator is stateless, thread-safe, and cacheable.
+type Validator interface {
+	// Validate validates the supplied value against the Validator CEL
+	// expression in context of the request.
+	// Returns 'true' if the value is valid (passes validation).
+	// Returned errors should be considered as internal/technical errors,
+	// and should NOT be returned unprocessed to end-users of the API.
```

Comment:
"unprocessed" is doing a lot of work here, but, it should (technically) be enough.

Reply:
I know, but this feature could be further improved. This PR should be considered an MVP.